python/mlx/optimizers/optimizers.py

```diff
@@ -736,3 +736,31 @@ def apply_single(self, gradient: mx.array, parameter: mx.array, state: dict):
         if self.weight_decay != 0:
             parameter += parameter * (-self.weight_decay * learning_rate)
         return parameter - update
+
+
+def clip_grad_norm(grads, max_norm):
+    """
+    Clips the gradients by scaling them when their norm exceeds a maximum threshold.
+
+    This function ensures that the norm of the gradients does not exceed `max_norm` to prevent issues like exploding gradients in neural networks. It scales down the gradients proportionally if their norm is greater than `max_norm`.
```

Comment:
I know it is a nitpick but since the rest of the docstrings are wrapped at something like 80 chars could you please wrap these docstrings as well?

Also the first sentence should start in the same line as `"""`.